src/cs/usfca/edu/histfavcheckout/controller/HistFavCheckoutHandler.java

```diff
@@ -79,6 +87,25 @@ public User addUser(PrimaryKey id) {
 		return ResponseEntity.status(HttpStatus.OK).body(products);		
 	}
 	
+	public ResponseEntity<?> getCheckouts(int userId, int page, int nums) {
+		List<User> userCheckedOutMovies = userRepository.findCheckedOutMovies(userId, true, 
+				PageRequest.of(page, nums, Sort.by("numberOfFavorites").descending()));
+		HashMap<Integer, User> movieMap = new HashMap();
+		for(User u: userCheckedOutMovies) {
+			movieMap.put(u.getId().getProductId(), u);
+		}
+		//TODO: call search team's API here
+		SearchMoviesResponse searchAPIResp = APIClient.getAllMovies(movieMap.keySet());
+		GetUserCheckoutsResponse checkouts = new GetUserCheckoutsResponse();
+		for(MovieData m : searchAPIResp.getResults()) {
+			User usr = movieMap.get(m.getID());
+			String checkoutDate = getCheckoutDate(usr.getExpectedReturnDate());
+			Movie mov = checkouts.new Movie(m.getTitle(), m.getID(), checkoutDate);
+			checkouts.addMovie(mov);
+		}
+		return ResponseEntity.status(HttpStatus.OK).body(checkouts);		
```

Comment:
Not sure if we should sort based on checkout date prior to returning. What do you think?